app/controllers/declaration/ItemsSummaryController.scala

```diff
@@ -154,11 +155,7 @@ class ItemsSummaryController @Inject() (
   private def removeItemFromCache(itemId: String)(implicit request: JourneyRequest[AnyContent]): Future[ExportsDeclaration] =
     request.cacheModel.itemBy(itemId) match {
       case Some(itemToDelete) =>
-        val updatedItems =
-          request.cacheModel.items.filterNot(_ == itemToDelete).zipWithIndex.map { case (item, index) =>
-            item.copy(sequenceId = index + 1)
-          }
-
+        val updatedItems = request.cacheModel.items.filterNot(_ == itemToDelete)
```

Comment:
As I assume the `==` op might be quite costly considering how nested is the `ExportItem` case class, wouldn't be just using the `id` field sufficient (and probably safer) for filtering out the instance to remove? WDYT?

Reply:
Yeah agree with that